hub/src/server/server.ts

```diff
@@ -83,24 +101,31 @@ export class HubServer {
 
     // can the caller delete? if so, in what paths?
     const scopes = getAuthenticationScopes(requestHeaders.authorization)
-    const deletePrefixes = []
-    const deletePaths = []
-    for (let i = 0; i < scopes.length; i++) {
-      if (scopes[i].scope == 'deleteFilePrefix') {
-        deletePrefixes.push(scopes[i].domain)
-      } else if (scopes[i].scope == 'deleteFile') {
-        deletePaths.push(scopes[i].domain)
+    const isArchivalRestricted = scopes.writeArchivalPaths.length > 0 || scopes.writeArchivalPrefixes.length > 0
```

Comment:
This code is duplicated below.  I think there could be a method that determines (1) whether or not we're authorized to write to this path and (2) if so, whether or not we're archival-restricted.

Reply:
DRY'd this up